### PR TITLE
Testing post service

### PR DESCRIPTION
**Pull request by kevinchu93**
add tests for getAllPosts and createNewPosts in PostService
@ericchu94

### CODE REVIEW

File: mongoose.js [hunk 1]
```diff
@@ -1,7 +1,3 @@
 const mongoose = require('mongoose');
```
Comment (ericchu94) [inline mongoose.js]: Do you need this file anymore?

File: test/postServiceTest.js [hunk 2]
```diff
@@ -6,17 +6,47 @@ const PostService = require('../services/PostService');
...
       assert.equal(post, expected);
       stub.restore();
     });
-    it('should call function findById with correct parameters', () => {
-      const stub = sinon.stub(Post, 'findById');
-      PostService.getOnePost(5);
-      sinon.assert.calledWith(stub, 5);
+  });
+  describe('getAllPosts()', () => {
+    it('should be called with no parameters', () => {
```
Comment (ericchu94) [inline test/postServiceTest.js]: once again, it refers to `getAllPosts()`, rather than `find()`.

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: app.js [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+
+mongoose.connect('mongodb://test:test@ds257627.mlab.com:57627/first-blog-mern', { useMongoClient: true });
+
 app.set('view engine', 'ejs');
 app.use(express.static('public'));
 
 app.use('', postController);
 
-app.listen(3000);
+module.exports = app.listen(3000);
```
Comment (ericchu94) [inline app.js]: What is this used for?

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: services/PostService.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ class PostService {
   }
 
   static createNewPost(title, author, body) {
-    return new Post({ title, author, body }).save();
+    return Post.create({ title, author, body });
```
Comment (ericchu94) [inline services/PostService.js:13]: This is good; it's common that `new` makes things harder to test

File: mongoose.js (hunk 1 above)
Comment (kevinchu93) [reply mongoose.js]: No, removed

File: test/postServiceTest.js (hunk 2 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: fixed

File: app.js (hunk 3 above)
Comment (kevinchu93) [reply app.js]: For my controller test. I missed this change when I was moving all controller test changes to its own branch. Removed.

File: services/PostService.js (hunk 4 above)
Comment (kevinchu93) [reply services/PostService.js:13]: Nice. I didn't know how to use Sinon's stub format of sinon.stub(object, method of object); if it was Post().save();

File: test/postServiceTest.js [hunk 5]
```diff
@@ -36,7 +36,7 @@ describe('PostService', () => {
     });
   });
   describe('createNewPost(title, author, body)', () => {
-    it('should be called with correct parameters', () => {
+    it('should call function createNewPost with correct parameters', () => {
```
Comment (ericchu94) [inline test/postServiceTest.js]: You aren't stubbing createNewPost, you are stubbing create.

**Review by ericchu94 — CHANGES_REQUESTED** [review]

File: test/postServiceTest.js (hunk 5 above)
Comment (kevinchu93) [reply test/postServiceTest.js]: Right, fixed.

**Review by ericchu94 — APPROVED** [review]

**Pull request merged**